app/v2/notifications/post_notifications.py

```diff
@@ -244,9 +249,32 @@ def process_document_uploads(personalisation_data, service, simulated=False):
             except DocumentDownloadError as e:
                 raise BadRequestError(message=e.message, status_code=e.status_code)
 
+    if not simulated:
+        save_stats_for_attachments(
+            [v for k, v in personalisation_data.items() if k in file_keys],
+            service.id,
+            template_id
+        )
+
     return personalisation_data
 
 
+def save_stats_for_attachments(files_data, service_id, template_id):
+    nb_files = len(files_data)
+    statsd_client.incr(f"attachments.nb-attachments.count-{nb_files}")
+    statsd_client.incr('attachments.nb-attachments', count=nb_files)
+    statsd_client.incr(f"attachments.services.{service_id}", count=nb_files)
+    statsd_client.incr(f"attachments.templates.{template_id}", count=nb_files)
+
+    for document in [f['document'] for f in files_data]:
```

Comment:
Is there a potential for raise `KeyError`, does the caller up to the REST endpoint able to handle this?

Reply:
No, a potential `KeyError` is not caught. If some keys are not there that's a real error. We control responses from the document download API and it should match the output from `document_download_response` in tests. Or you can look at the document API HTTP endpoint as well

Reply:
We've got a similar assumption about keys here as well. This is covered by tests

https://github.com/cds-snc/notification-api/blob/46f9ce23d0acc8ff52fb65fe96ac72ea3f5a40bc/app/delivery/send_to_providers.py#L108-L142